felix/dataplane/linux/int_dataplane.go

```diff
@@ -918,20 +924,73 @@ func ConfigureDefaultMTUs(hostMTU int, c *Config) {
 	}
 }
 
+func cleanUpIPIPAddrs() {
+	// If IPIP is not enabled, check to see if there is are addresses in the IPIP device and delete them if there are.
```

Comment:
I think we need to be careful when cleaning up the IPIP tunnel device.  `tunl0` is a shared resource, it doesn't belong to Calico.  With our IPIP disabled, the user should be able to add their own IP to tunl0 without us removing it.

I'm aware of several users running Calico in VXLAN mode so they can use IPIP for something else.

Reply:
I just pushed changes to use the previously retrieved IP pool list and only attempt to remove addresses that belong to a Calico IP pool. What do you think about this approach? Would just leaving the addresses there be more desirable than this?
Thanks!

Reply:
> I just pushed changes to use the previously retrieved IP pool list and only attempt to remove addresses that belong to a Calico IP pool. 

This is an OK but imperfect fix. One example being the user deletes the IPPool that enables IPIP, and thus we simultaneously see that IPIP is no longer enabled, but also see that the addresses assigned to the tunnel are not within an IP pool.

I'm not sure if there's a perfect answer here - we either need to leave the addresses on the tunnel and suffer IP duplication in the network, or potentially delete other users IPs.

The best I can come up with here is this:
- If IPIP is explicitly disabled, then don't clean up the interface at all.
- If IPIP is implicitly disabled, clean up the interface.
- If IPIP is explicitly or implicitly enabled, existing logic kicks in.

That way, users who are applying their own addresses can at least tell Felix not to touch their IPs by explicitly disabling IPIP via FelixConfiguration. WDYT?

Reply:
Sounds good! Yes, 100% agree using the pool list is imperfect (I even added a comment saying so on L963-966), this works way better. Thanks, will do this and push.